src/Pagination.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import BootstrapMixin from './BootstrapMixin';
+
+const Pagination = React.createClass({
+  mixins: [BootstrapMixin],
+
+  propTypes: {
+    activePage: React.PropTypes.number,
+    items: React.PropTypes.number,
+    maxButtons: React.PropTypes.number,
+    ellipsis: React.PropTypes.bool,
+    first: React.PropTypes.bool,
+    last: React.PropTypes.bool,
+    prev: React.PropTypes.bool,
+    next: React.PropTypes.bool,
+    onSelect: React.PropTypes.func
+  },
+
+  getDefaultProps() {
+    return {
+      activePage: 1,
+      items: 1,
+      maxButtons: null,
+      first: false,
+      last: false,
+      prev: false,
+      next: false,
+      ellipsis: true,
+      bsClass: 'pagination'
+    };
+  },
+
+  handleSelect(pagenumber) {
+    if(this.props.onSelect){
+      this.props.onSelect(pagenumber);
+    }
+  },
+
+  renderButton({child, key, className, onClick}){
+    //TODO: replace <a> with <SafeAnchor> when it is avaliable
+    return (
+      <li
+        key={key}
+        onClick={onClick}
+        className={className}>
+        <a href='#' onClick={e => e.preventDefault()}>{child}</a>
+      </li>
+    );
```

Comment:
I think the way to combine that `onClick` event would be to pull this into a separate component. something like:

```
class PaginationButton extends React.Component {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(event) {
    // This would go away once SafeAnchor is available
    event.preventDefault();

    if (this.props.onSelect) {
      let selectEvent = createSelectEvent(this.props.eventKey);
      this.props.onSelect(event, selectEvent);
    }
  }

  render() {
    return (
      <li className={className}>
        <a href='#' onClick={handleClick}>{child}</a>
      </li>
    );
  }
}

PaginationButton.PropTypes = {
  eventKey: React.PropTypes.oneOfType([
    React.PropTypes.string,
    React.PropTypes.number
  ]),
  onSelect: React.PropTypes.func
};
```

Some of these concepts are getting inline with #419 and would be easier with #787 which is what I've been working on with the dropdown-revisited work.